homeassistant/components/sensor/__init__.py

```diff
@@ -141,7 +149,100 @@ def capability_attributes(self) -> Mapping[str, Any] | None:
     @property
     def state_attributes(self) -> dict[str, Any] | None:
         """Return state attributes."""
+        attrs: dict[str, Any] = {}
         if last_reset := self.last_reset:
-            return {ATTR_LAST_RESET: last_reset.isoformat()}
+            attrs[ATTR_LAST_RESET] = last_reset.isoformat()
+        if native_value := self.native_value:
+            attrs[ATTR_NATIVE_VALUE] = native_value
```

Comment:
I don't really like that we're polluting the state attributes with the native value, but I don't have a better idea.

Reply:
We _could_ avoid setting the `native_value` attribute if it's equal to the `state` and also avoid setting the `native_unit_of_measurement` if it's equal to `unit_of_measurement`
However, maybe it's just confusing if the attributes are there sometimes?

Reply:
I don't think we should add this now. This PR is not changing the last state behavior. This problem was present before and integration had to deal with it then too.